### PR TITLE
fix: Remove/Disable Duplicate button in Read Only mode of an event-type

### LINKED ISSUE #15509: [CAL-3959] Remove/Disable Duplicate button in Read Only mode of an event-type
Add a user to a team with MEMBER role. He can't create any event for that team. But for an existing event he sees duplicate button which errors in the end.

image

<sub>[CAL-3959](https://linear.app/calcom/issue/CAL-3959/removedisable-duplicate-button-in-read-only-mode-of-an-event-type)</sub>

### PR DESCRIPTION
**Pull request by Amit91848**
## What does this PR do?

- Fixes #15509 
- Fixes CAL-3959

Screenshot 2024-06-20 162705

## Mandatory Tasks (DO NOT REMOVE)

- [x] I have self-reviewed the code (A decent size PR without self-review might be rejected)
- [x] N/A I have added a Docs issue [here](https://github.com/calcom/docs/issues/new) if this PR makes changes that would require a [documentation change](https://docs.cal.com)
- [x] I have added or modified automated tests that prove my fix is effective or that my feature works (PRs might be rejected if logical changes are not properly tested)

### CODE REVIEW

File: apps/web/modules/event-types/views/event-types-listing-view.tsx [hunk 1]
```diff
@@ -513,19 +513,22 @@ export const EventTypeList = ({
...
-                                      data-testid={`event-type-duplicate-${type.id}`}
-                                      StartIcon="copy"
-                                      onClick={() => openDuplicateModal(type, group)}>
-                                      {t("duplicate")}
-                                    </DropdownItem>
-                                  </DropdownMenuItem>
-                                </>
-                              )}
+                              {/* readonly is only set when we are on a team - if we are on a user event type null will be the value. */}
+                              {(group.metadata?.readOnly === false || group.metadata.readOnly === null) &&
```
Comment (ThyMinimalDev) [inline apps/web/modules/event-types/views/event-types-listing-view.tsx:517]: does checking the role of the member withing the organization makes sense here ?

Comment (Amit91848) [reply apps/web/modules/event-types/views/event-types-listing-view.tsx]: Switched to use readOnly variable which is also being used in other places.

**Review by anikdhabal — CHANGES_REQUESTED** [review]
Hey @Amit91848, please remove the duplicate button for small screens as well.

Can we update the duplicate event type test to ensures that the readonly badge is not visible before proceeding to duplicate the event type?

**Review by anikdhabal — APPROVED** [review]
LGTM

Updated the test case

**Pull request merged**